hi-ml-cpath/src/health_cpath/utils/montage.py

```diff
@@ -0,0 +1,517 @@
+            print("Inclusion list will be matched against the Slide ID column (for predefined datasets) or the "
+                  "filename.")
+            return slides_to_include
+        else:
+            return []
+
+    def read_dataset(self, input_folder: Path) -> DatasetOrDataframe:
+        """Read the dataset that should be used for creating the montage. If the dataset is recognized as either
+        TCGA-PRAD or TCGA-BRCA, the dataset will be read from the predefined dataset. Otherwise, all image files in the
+        input folder will be used.
+
+        :param input_folder: The folder where the dataset is located.
+        :return: A SlidesDataset or dataframe object that contains the dataset."""
+        if self.image_glob_pattern:
+            print(f"Trying to create a dataset from files in the input folder that match: {self.image_glob_pattern}")
+            try:
+                dataset = dataset_from_folder(input_folder, glob_pattern=self.image_glob_pattern)
+            except Exception as ex:
+                raise ValueError(f"Unable to create dataset from files in folder {input_folder}: {ex}")
+            if len(dataset) == 0:
+                raise ValueError(f"No images found in folder {input_folder} with pattern {self.image_glob_pattern}")
+            return dataset
+        else:
+            csv_file = input_folder / SlidesDataset.DEFAULT_CSV_FILENAME
+            print(f"Trying to load the dataset as a SlidesDataset object from {SlidesDataset.DEFAULT_CSV_FILENAME}")
+            if csv_file.is_file():
+                return SlidesDataset(root=input_folder)
+            raise ValueError(
+                f"No dataset file '{SlidesDataset.DEFAULT_CSV_FILENAME}' found in the input folder. Please add "
+                "this file, or provide a glob pattern to find the slides for the montage via --image_glob_pattern"
+            )
+
+    def create_montage(self, input_folder: Path) -> None:
+        dataset = self.read_dataset(input_folder)
+        self.output_path.mkdir(parents=True, exist_ok=True)
+        if self.include_by_slide_id and self.exclude_by_slide_id:
+            raise ValueError("You cannot provide both an inclusion and exclusion list.")
+        if self.include_by_slide_id:
+            items = self.read_inclusion_list()
+            exclude_items = False
+        elif self.exclude_by_slide_id:
+            items = self.read_exclusion_list()
+            exclude_items = True
+        else:
+            items = []
+            exclude_items = True
+        montage_from_included_and_excluded_slides(
+            dataset=dataset,
+            items=items,
+            exclude_items=exclude_items,
+            output_path=self.output_path,
+            width=self.width,
+            num_parallel=self.parallel,
+            backend=self.backend,
+        )
+
+
+def create_config_from_args() -> MontageConfig:
```

Comment:
I've moved the parameters that we will need to re-use into a separate class. I've not renamed this function though because from its module name it is clear that it is only about montages.